src/functional/ffront/past_passes/type_deduction.py

```diff
@@ -69,9 +69,14 @@ def _validate_call_params(new_func: past.Name, new_kwargs: dict):
                 raise GTTypeError(
                     f"Only 2 values allowed in domain range, but got `{len(domain_values.elts)}`."
                 )
-            if domain_values.elts[0].type != ct.ScalarType(
-                kind=ct.ScalarKind.INT64
-            ) or domain_values.elts[1].type != ct.ScalarType(kind=ct.ScalarKind.INT64):
+            scalar_types_check = [
+                ct.ScalarType(kind=ct.ScalarKind.INT64),
+                ct.ScalarType(kind=ct.ScalarKind.INT32),
+            ]
+            if (
+                domain_values.elts[0].type not in scalar_types_check
+                or domain_values.elts[1].type not in scalar_types_check
+            ):
```

Comment:
I think it's worth adding a small utility function in `type_info.py`. There is already `is_integral` but that would also be true for integral fields. Therefore I propose to add `is_integral_scalar`, then the code here would be much easier to read:
```
if not is_inegral_scalar(domain_values.elts[0].type) or not is_integral_scalar(domain_values.elts[1].type)
```
Also you would not forget to add `ScalarKind.INT` to the list.